lua/cfc_random_spawn/module/sv_player_spawning.lua

```diff
@@ -105,22 +103,43 @@ local function getLivingPlayers()
     return livingPlayers
 end
 
--- Get the first SELECTION_SIZE spawns that are closest to nearPos and are within range of CENTER_CUTOFF_SQR
-local function getNearestSpawns( nearPos, spawns )
-    local tempDistanceTable = {}
+local function getNearestSpawn( nearPos, spawns )
+    local nearestSpawn
+    local closestDistSqr = math.huge
+    for _, spawn in ipairs( spawns ) do
+        local distToNearSqr = spawn.spawnPos:DistToSqr( nearPos )
+        if distToNearSqr < closestDistSqr then
+            closestDistSqr = distToNearSqr
+            nearestSpawn = spawn
+        end
+    end
+
+    if not nearestSpawn then return spawns[1] end
+
+    return nearestSpawn
```

Comment:
you could probably  delete this because you should always have a nearestSpawn set starting with math.huge
```suggestion
    return nearestSpawn
```

Reply:
👍